maxfw/tests/test_image_utils.py

```diff
@@ -0,0 +1,171 @@
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.array(img_out).shape == (200, 200, 4)
+
+    # Resize to 2000x2000
+    transform_sequence = [ToPILImage('RGBA'), Resize(size=(2000, 2000))]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.array(img_out).shape == (2000, 2000, 4)
+
+
+def test_imageprocessor_grayscale():
+    """Test the Imageprocessor's grayscale function."""
+
+    # Using the standard 1 output channel
+    transform_sequence = [ToPILImage('RGBA'), Resize(size=(200, 200)), Grayscale()]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.array(img_out).shape == (200, 200)
+
+    # Using 3 output channels
+    transform_sequence = [ToPILImage('RGBA'), Resize(size=(200, 200)), Grayscale(num_output_channels=3)]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.array(img_out).shape == (200, 200, 3)
+
+    # Using 4 output channels
+    transform_sequence = [ToPILImage('RGBA'), Resize(size=(200, 200)), Grayscale(num_output_channels=4)]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.array(img_out).shape == (200, 200, 4)
+
+
+def test_imageprocessor_normalize():
+    """Test the Imageprocessor's normalize function."""
+
+    # Test normalize
+    transform_sequence = [ToPILImage('RGBA'), Normalize()]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.max(img_out) <= 1 and np.min(img_out) >= 0
+
+    # Test normalize
+    transform_sequence = [ToPILImage('RGB'), Normalize(), Normalize()]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.max(img_out) <= 1 and np.min(img_out) >= 0
+
+    # Test normalize
+    transform_sequence = [ToPILImage('L'), Normalize()]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.max(img_out) <= 1 and np.min(img_out) >= 0
+
+
+def test_imageprocessor_standardize():
+    """Test the Imageprocessor's standardize function."""
+
+    # Test standardize
+    transform_sequence = [ToPILImage('RGBA'), Standardize()]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert round(np.std(img_out)) == 1
+
+    # Test standardize
+    transform_sequence = [ToPILImage('RGB'), Standardize(), Standardize()]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert round(np.std(img_out)) == 1
+
+    # Test standardize
+    transform_sequence = [ToPILImage('L'), Standardize()]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert round(np.std(img_out)) == 1
+
+
+def test_imageprocessor_rotate():
```

Comment:
Would be good to expand test cases here to test actual values (like the operator gives the same output as the underlying PIL function)

Reply:
Thanks Nick, have added this to the next update.